crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -244,13 +254,53 @@ pub trait PositionManager: StateWrite + PositionRead {
         self.vcb_credit(position.reserves_1()).await?;
         self.vcb_credit(position.reserves_2()).await?;
 
+        // Add the asset IDs from the new position's trading pair
+        // to the candidate set for this block.
+        let routing_params = self.routing_params().await?;
+        self.add_recently_accessed_asset(
+            position.phi.pair.asset_1(),
+            routing_params.fixed_candidates.clone(),
+        );
+        self.add_recently_accessed_asset(
+            position.phi.pair.asset_2(),
+            routing_params.fixed_candidates,
```

Comment:
It was a minor surprise that this wasn't internalized by `add_recently_accessed_asset` but I guess we technically save a lookup.